.buildkite/custom-tests.json

```diff
@@ -20,7 +20,7 @@
     },
     {
       "test_name": "staging: style",
-      "command": "cd staging && cargo fmt --all -- --check --config format_code_in_doc_comments=true"
+      "command": "cd staging && cargo +nightly fmt --all -- --config-path rustfmt.toml --check"
```

Comment:
Should we maybe ping the nightly version number down to prevent a nightly update breaking CI? Or does that already happen elsewhere?

Would it make sense to add an overview of which of the format options are nightly only? Eventually, we would probably want to drop the nightly dependency, right?

Reply:
I also suggest to leave `custom-tests.json` as much as possible equal to the one used out of staging (maybe we should change the name).

So my suggestion is to add another file with change, where we can also run it on no-staging crates. WDYT?

Reply:
Yes we do want to drop it, I am not familiar with how rustfmt rules get stable though. What would be the best approach you think?

Reply:
@stefano-garzarella a separate `.json` file with just the nightly fmt? is that possible?

Reply:
@stefano-garzarella there are too many formatting changes 🤔 It'd cause a big diff upheaval 😀

Reply:
Here's the result on `vhost-device`: 

```shell
epilys /rust-vmm/vhost-device % cargo +nightly fmt --all -- --config-path staging/rustfmt.toml
epilys /rust-vmm/vhost-device % gs
On branch rustfmt.toml
Your branch is up to date with 'origin/rustfmt.toml'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   vhost-device-gpio/src/backend.rs
        modified:   vhost-device-gpio/src/gpio.rs
        modified:   vhost-device-gpio/src/mock_gpio.rs
        modified:   vhost-device-gpio/src/vhu_gpio.rs
        modified:   vhost-device-i2c/src/i2c.rs
        modified:   vhost-device-i2c/src/main.rs
        modified:   vhost-device-i2c/src/vhu_i2c.rs
        modified:   vhost-device-rng/src/main.rs
        modified:   vhost-device-rng/src/vhu_rng.rs
        modified:   vhost-device-scmi/src/devices/common.rs
        modified:   vhost-device-scmi/src/devices/fake.rs
        modified:   vhost-device-scmi/src/devices/iio.rs
        modified:   vhost-device-scmi/src/devices/mod.rs
        modified:   vhost-device-scmi/src/main.rs
        modified:   vhost-device-scmi/src/scmi.rs
        modified:   vhost-device-scmi/src/vhu_scmi.rs
        modified:   vhost-device-scsi/src/main.rs
        modified:   vhost-device-scsi/src/scsi/emulation/block_device.rs
        modified:   vhost-device-scsi/src/scsi/emulation/command.rs
        modified:   vhost-device-scsi/src/scsi/emulation/missing_lun.rs
        modified:   vhost-device-scsi/src/scsi/emulation/response_data.rs
        modified:   vhost-device-scsi/src/scsi/emulation/target.rs
        modified:   vhost-device-scsi/src/scsi/emulation/tests/bad_lun.rs
        modified:   vhost-device-scsi/src/scsi/emulation/tests/generic.rs
        modified:   vhost-device-scsi/src/scsi/emulation/tests/mod.rs
        modified:   vhost-device-scsi/src/scsi/emulation/tests/report_supported_operation_codes.rs
        modified:   vhost-device-scsi/src/scsi/sense.rs
        modified:   vhost-device-scsi/src/vhu_scsi.rs
        modified:   vhost-device-scsi/src/virtio.rs
        modified:   vhost-device-vsock/src/main.rs
        modified:   vhost-device-vsock/src/thread_backend.rs
        modified:   vhost-device-vsock/src/txbuf.rs
        modified:   vhost-device-vsock/src/vhu_vsock.rs
        modified:   vhost-device-vsock/src/vhu_vsock_thread.rs
        modified:   vhost-device-vsock/src/vsock_conn.rs

no changes added to commit (use "git add" and/or "git commit -a")
```

Reply:
`35 files changed, 664 insertions(+), 503 deletions(-)`

Reply:
If we add that commit to `.git-blame-ignore-revs` file, what could be other issues?

(Anyway, for now I'm fine to enable it only for staging)

Reply:
My approach would be to gradually lift these rules everywhere. For example it was in `sound` only at first. Now it's `staging`. Then the root workspace in a follow up PR.

Reply:
We can do one commit per rule in root crates then, and at the end add the rustfmt.toml file for all crates. What do you think?

Reply:
@stefano-garzarella

> If we add that commit to `.git-blame-ignore-revs` file, what could be other issues?
> 
> (Anyway, for now I'm fine to enable it only for staging)

None, just the amount of lines to review at once. Sometimes rustfmt has bugs and we should be on the lookout in general.

Reply:
> My approach would be to gradually lift these rules everywhere. For example it was in `sound` only at first. Now it's `staging`. Then the root workspace in a follow up PR.

Yep, I agree on that!
I'll send a PR to change `custom-tests.json` to `staging-tests.json` with a comment to explain that we want to leave it aligned with the `rust-vmm-ci` tests.

Reply:
> If we add that commit to `.git-blame-ignore-revs` file, what could be other issues?

I think that file is still opt-in, or did something change?

Reply:
> We can do one commit per rule in root crates then, and at the end add the rustfmt.toml file for all crates. What do you think?

Yep, I agree on that. 1 commit per rule with the parameters described in the commit description.

Reply:
> We can do one commit per rule in root crates then, and at the end add the rustfmt.toml file for all crates. What do you think?

Sounds good to me. Except, I would probably add the rules in some rust-vmm wide manner.